DashAI/back/api/api_v1/endpoints/components.py

```diff
@@ -1,37 +1,66 @@
+"""Component API module."""
 import logging
+from typing import Annotated, Any
 
-from fastapi import APIRouter, status
+from fastapi import APIRouter, Query, status
 from fastapi.exceptions import HTTPException
 
-from DashAI.back.core.config import name_registry_mapping
+from DashAI.back.core.config import component_registry
 
 logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 router = APIRouter()
 
 
+def _intersect_component_lists(
+    previous_selected_components: dict[str, dict[str, Any]],
+    component_list: dict[str, dict[str, Any]],
+) -> dict[str, dict[str, Any]]:
+    selected_components = {
+        component_dict["name"]: component_dict
+        for component_dict in component_list
+        if component_dict["name"] in previous_selected_components
+    }
+    return selected_components
+
+
+def _delete_class(component_dict: dict[str, Any]) -> dict[str, Any]:
+    return {key: value for key, value in component_dict.items() if key != "class"}
+
+
 @router.get("/")
 async def get_components(
-    component_type: str | None = None,
-    task_name: str | None = None,
+    select_types: Annotated[list[str] | None, Query()] = None,
+    ignore_types: Annotated[list[str] | None, Query()] = None,
+    related_component: str | None = None,
     component_parent: str | None = None,
 ) -> list[dict]:
-    """Retrieves components from the register.
+    """Retrieve components from the register according to the provided parameters.
+
+    When all parameters are None, the method return all registered components.
+    If more than one parameter is specified, then the method returns the intersection
+    between the retrived components.
 
     The components returned will depend on the parameters of the request.
-    If all parameters are None, then the method will return all registered components.
+    If all parameters are None, then the method returns all registered components.
 
     Parameters
     ----------
-    component_type : str | None, optional
-        If specified, the function will return only the components belonging to that
-        component type (e.g., task, model, dataloader, etc...), by default None.
-    task_name : str | None, optional
-        If specified, the function will return only the components related with
-        the task (e.g., TabularClassification, Translation), by default None.
+    select_types: Annotated[list[str] | None, Query()], optional
```

Comment:
Just to know, why did you change this name? It sounded more natural to me as component_types

Reply:
I changed it because I also added the option to ignore types.
So, the select was more explicit to say that I want to select only certain types.